### PR TITLE
Feature/edit questionnaire

### PR DESCRIPTION
**Pull request by i-am-dan**
@hls-hack 

Please take a look!

Questionnaire page > Edit button should go to edit page with questions prepopulated with texts.
Patient List page > Edit button should be hidden when selecting a questionnaire.

### CODE REVIEW

**Review by ansgoyal — APPROVED** [review]

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿@page "/questionnaire/edit/{id}"
+using FhirBlaze.SharedComponents
+@using Hl7.Fhir.Model
+@using Microsoft.AspNetCore.Components.Forms
+@using FhirBlaze.QuestionnaireModule.Components
+
+@if(!QLoaded){
+    <h4>Loading</h4>
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor:8]: Replace with loading spinners

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}else{
+    <h4>@Questionnaire.Description</h4>
+    <h5>(ID: @Questionnaire.Id)</h5>
+}
+@if (!QLoaded)
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor:13]: please combine this with the previous if statement

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                Questionnaire = await FhirService.GetQuestionnaireByIdAsync(Id);
+                QLoaded = true;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.ToString());
+            }
+        }       
+
+        protected void AddItem(QuestionnaireItemType type)
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs:42]: This method is not used; please remove

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace FhirBlaze.QuestionnaireModule.Pages
+{
+    [Authorize]
+    public partial class EditQuestionnaire
+    {
+        [Inject]
+        public IFhirService FhirService { get; set; }
+        [Inject]
+        public HttpClient Http { get; set; }
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs:21]: HttpClient is never used

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        }       
+
+        protected void AddItem(QuestionnaireItemType type)
+        {
+            var nItem = new Questionnaire.ItemComponent();
+            nItem.Type = type;
+            Questionnaire.Item.Add(nItem);
+        }
+
+        protected void RemoveItem(Questionnaire.ItemComponent Item)
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs:49]: Not used

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            Questionnaire.Item.Remove(Item);
+        }
+
+        protected string GetHeader(Questionnaire.ItemComponent item)
```
Comment (SameerDoshi) [inline FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs:54]: Not used

**Review by SameerDoshi — CHANGES_REQUESTED** [review]
Please make minor updates to remove unused code

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor (hunk 1 above)
Comment (i-am-dan) [reply FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor:8]: done

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor (hunk 2 above)
Comment (i-am-dan) [reply FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor:13]: Fixed

File: FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs (hunk 3 above)
Comment (i-am-dan) [reply FhirBlaze.QuestionnaireModule/Pages/EditQuestionnaire.razor.cs:42]: This was used before I added the questionnaire component.

**Review by SameerDoshi — APPROVED** [review]

**Pull request merged**